### PR TITLE
EVG-5181 Use project not task queue to find tasks to block

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/task_queue.go [hunk 1]
```diff
@@ -155,34 +155,44 @@ func ValidateNewGraph(t *task.Task, tasksToBlock []string) error {
...
+			if err != nil {
+				catcher.Add(errors.Wrapf(err, "problem finding task %s", tgTask))
+			}
+			if found == nil {
+				catcher.Add(errors.Errorf("found nil task %s", tgTask))
+				continue
+			}
+			tasksToBlock = append(tasksToBlock, *found)
+		}
+		if t.DisplayName == tgTask {
```
Comment (john-m-liu) [inline model/task_queue.go]: If you find the task I think you want to have every task in the group in tasksToBlock right? Right now you're adding a potentially random subset of them

File: model/task/db.go [hunk 2]
```diff
@@ -221,6 +221,14 @@ func ByVersionsForNameAndVariant(versions []string, displayName, buildVariant st
 	})
 }
 
+func ByVersionForNameAndVariant(version string, displayName, buildVariant string) db.Q {
```
[the remark this answers is not in the document]
Comment (bsamek) [reply model/task/db.go]: Fixed.

[the remark this answers is not in the document]
Comment (tychoish) [reply model/task/db.go]: Can we make new queries be functions that take arguments and return results. 

Passing this to findOne is potentially weird given that there's isn't a sort, but maybe there are (effectively) unique  documents.

Additionally you only need to specify the type name ones in the function signature, which is itself a kind of a problem, given that it's easy to twiddle three-successive IDs

File: model/task_queue.go [hunk 3]
```diff
@@ -155,34 +155,44 @@ func ValidateNewGraph(t *task.Task, tasksToBlock []string) error {
...
 		return errors.Errorf("found nil task %s", taskID)
 	}
+
+	p, err := FindProjectFromTask(t)
+	if err != nil {
+		return errors.Wrapf(err, "problem getting project for task %s", t.Id)
+	}
+	tg := p.FindTaskGroup(t.TaskGroup)
+	encounteredTask := false
+	for _, tgTask := range tg.Tasks {
```
Comment (tychoish) [inline model/task_queue.go]: is it possible to express this logic as an aggregation operation...

File: model/task_queue.go (hunk 1 above)
Comment (bsamek) [reply model/task_queue.go]: I want all later tasks to be blocked. Task group tasks in a single-host task group are dispatched in the order defined in the configuration file. The idea here is to iterate over the tasks in the task group until we find the one that failed, then block all later tasks.

File: model/task_queue.go (hunk 3 above)
Comment (bsamek) [reply model/task_queue.go]: This logic is fairly complex. I'm saying, given an array like `["a", "b", "c", "d", "e"]` and a string like `"c"`, construct a query from all array elements with indexes greater than `"c"`'s.

**Pull request merged**